src/rpdk/core/data_loaders.py

```diff
@@ -158,13 +158,17 @@ def load_resource_spec(resource_spec_file):  # pylint: disable=R0912 # noqa: C90
             list_options,
         )
 
-    if set(resource_spec.get("readOnlyProperties", [])) & set(
-        resource_spec.get("createOnlyProperties", [])
-    ) or set(resource_spec.get("readOnlyProperties", [])) & set(
-        resource_spec.get("writeOnlyProperties", [])
-    ):
+    read_only_properties_intersection = set(
+        resource_spec.get("readOnlyProperties", [])
+    ) & (
+        set(resource_spec.get("createOnlyProperties", []))
+        | set(resource_spec.get("writeOnlyProperties", []))
+        | {"/properties/" + s for s in resource_spec.get("required", [])}
```

Comment:
think this wouldn't catch nested `required` properties overlapping with [`readOnlyProperties`](https://github.com/aws-cloudformation/aws-cloudformation-resource-schema/#resource-semantics)